### ISSUE TITLE
Lost comment on network disconnection

### ISSUE THREAD

**Issue by vlotorev** [jruesga/rview#52]
Use case:
* start edit review comment
* network connection is lost, user is not aware of this
* click Save button, but 'Server instance cannot be reached' or 'No network connectivity detected' message appears.
* comment is lost.

Expected behaviour:
* comment is not lost, it's still possible to copy and save it somewhere (or even app caches it locally and marks it so that its possible to post it later)

**Comment by jruesga** [comment]
All the app dialogs behaves the same way: the dialogs only capture the information needed to call the api, but they don't call the api itself. Things need to be rethought in all the order to allow this king of feature.

**Comment by vlotorev** [comment]
I don't agree this is not a bug. User's comments are lost silently. IMO, at least it should be possible to notify user about not being able to save the comment and still give user an ability to copy the message to another editor.

Editing comments during code review takes a lot of time, as it requires thoughtful reading and code analyzing. It's quite frustrating for user to lose comments **silently**.

**Comment by jruesga** [comment]
@vlotorev, I'm also agree that is a bug (sorry if the new categorization lead you to understood is wasn't), and I understood your problem and frustation. What I'm trying to explain with the new label is that is not a simple change it could be done with a few lines of code. Unlike issue #51, this bug affects to all the dialog of the app (rebase, assigne, abandone, ...), and it should be treated in a same way (I could probably easily fix what you request in this issue, but it should not be a good way to fix it). The main problem is that the dialog (the input information) is separate from the operation (the api call), and the dialog was destroyed before the api was call, so the dialog cannot determine if the api call will be fail (it just collect the user information). If I save the data from one dialog, I should do it for every single dialog of the app, and IMO copy the message and recreate the dialog is just not a good UX. It should probably move all the api calls into a SyncAdapter, which performs the operations in background, but it could be a nightmare of sync operations between different accounts, server/client state syncyng, ..., something that easy could require a change of more than 5000+ lines (hence the feature tag). That's what I said it require a whole rethinking.

In the other hand, it is also more uncommon than issue #51, and it doesn't silently made your comment lost like issue #51 (user got a message about what is what wrong with the operation), and I should disagree a bit that an inline comment should take a lot time (at least most of the times). It's just a comment over one line, and it is send to server just when save button is tapping, and once in the server is not lost.

Summarizing, what in trying to explain is that I need a bit of time to try to fix this bug :)

**Comment by vlotorev** [comment]
Great, thanks for a reply.
